security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -228,19 +271,24 @@ func toEnvoySecret(s *security.SecretItem, caRootPath string) *tls.Secret {
 			},
 		}
 	} else {
-		secret.Type = &tls.Secret_TlsCertificate{
-			TlsCertificate: &tls.TlsCertificate{
-				CertificateChain: &core.DataSource{
-					Specifier: &core.DataSource_InlineBytes{
-						InlineBytes: s.CertificateChain,
+		res := toEnvoySecretWithPrivateKeyProvider(s, pkpConf)
+		if res != nil {
```

Comment:
nit, a little bit more go idiom if `res` is not used beyond the if-statement:
```
if res := toEnvoySecretWithPrivateKeyProvider(s, pkpConf); res != nil {
 ...
} else {
 ...
}
```

Reply:
Done